src/commands/upload.ts

```diff
@@ -0,0 +1,300 @@
+    };
+    const context = vm.createContext(sandbox);
+    script.runInContext(context);
+
+    if (!sandbox.module || typeof sandbox.module.exports === "undefined") {
+      throw new Error("Failed to extract configuration from hardhat.config.js");
+    }
+    hardhatConfig = sandbox.module.exports;
+  } catch (error) {
+    throw new Error(`Error loading hardhat config: ${error}`);
+  }
+
+  // Extract required fields
+  const sourcesPath = hardhatConfig.paths?.sources;
+  const artifactsPath = hardhatConfig.paths?.artifacts;
+  const solidity = hardhatConfig.solidity;
+
+  let solidityVersion: string;
+  let optimizerEnabled = false;
+  let optimizerRuns = 0;
+
+  if (typeof solidity === "string") {
+    solidityVersion = solidity;
+  } else {
+    solidityVersion = solidity.version;
+    optimizerEnabled = solidity.settings?.optimizer?.enabled;
+    optimizerRuns = solidity.settings?.optimizer?.runs;
+  }
+
+  if (!sourcesPath || !artifactsPath || !solidityVersion) {
+    throw new Error("Unable to find required fields in hardhat configuration.");
+  }
+
+  return {
+    sourcesPath,
+    artifactsPath,
+    solidityVersion,
+    optimizerEnabled,
+    optimizerRuns,
+  };
+};
+
+const main = async (options: any) => {
+  const projectPath = path.resolve(options.path || process.cwd());
+
+  console.log(`started the uploading process for ${projectPath}`);
+
+  let rootContractName = options.contract;
+
+  try {
+    if (!rootContractName) {
+      rootContractName = getRootContractNameFromConfig(projectPath);
+    }
+
+    const {
+      sourcesPath,
+      artifactsPath,
+      solidityVersion,
+      optimizerEnabled,
+      optimizerRuns,
+    } = await readHardhatFile(projectPath);
+
+    // check if artifactsPath exists or not (if not means it's not compiled yet)
+    if (!fs.existsSync(artifactsPath)) {
+      throw new Error(
+        `The project is not compiled yet, pls run \`bunzz build\` to compile it.`
+      );
+    }
+
+    const solFiles = await getAllSolidityFiles(sourcesPath, projectPath);
+
+    if (!rootContractName) {
+      // get all the contracts from the artifactsPath
+      let contractNames: string[] = [];
+      for (const file of solFiles) {
+        contractNames = contractNames.concat(
+          await getContractNames(file.content)
+        );
+      }
+
+      if (contractNames.length === 0) {
+        throw new Error("No contracts found.");
+      }
+
+      // ask user to select the base contract if the rootContractName is empty in interactive mode
+      rootContractName = await askUserToSelectContract(contractNames);
+      console.log(`Selected Contract: ${rootContractName}`);
+    }
+
+    // get the artifacts of base contract
+    const { ABI, bytecode } = getArtifacts(projectPath, rootContractName);
+
+    // validate the artifacts and respective code
+
+    // store the artifacts and related infos to the bunzz server
+    console.log("abi:", ABI);
+    console.log("bytecode:", bytecode);
```

Comment:
Maybe we should remove these in the future, you can keep them for now since it's not a PR into `main`

Reply:
yeah, later we'll remove those and use for sending to bunzz server purpose